qcengine/programs/base.py

```diff
@@ -34,6 +34,16 @@ def register_program(entry_point: 'ProgramHarness') -> None:
     programs[name.lower()] = entry_point
 
 
+def unregister_program(name: str) -> None:
+    """
+    Unregisters a given program.
+    """
+
+    ret = programs.pop(name.lower(), None)
+    if ret is None:
+        raise KeyError(f"Program {name} is not registered with QCEngine")
```

Comment:
Is a raise really required here? If a user wants to unregister and its not there, is there any harm? I could see leaving this to be explicit about whats present and not.

Reply:
After discussion, I think leaving this is fine so it catches things like spelling errors and symmetric with the `register` fuction.